src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -51,59 +40,46 @@ internal class WorkloadListCommand : WorkloadCommandBase
 
-            _dotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
-            ReleaseVersion currentSdkReleaseVersion = new(currentSdkVersion ?? Product.Version);
-            _currentSdkFeatureBand = new SdkFeatureBand(currentSdkReleaseVersion);
-            
             _includePreviews = result.GetValueForOption(WorkloadListCommandParser.IncludePreviewsOption);
-            _tempDirPath = tempDirPath ??
-                           (string.IsNullOrWhiteSpace(
-                               result.GetValueForOption(WorkloadListCommandParser.TempDirOption))
-                               ? Path.GetTempPath()
-                               : result.GetValueForOption(WorkloadListCommandParser.TempDirOption));
-            _targetSdkVersion = result.GetValueForOption(WorkloadListCommandParser.VersionOption);
-            _userProfileDir = userProfileDir ?? CliFolderPathCalculator.DotnetUserProfileFolderPath;
-            var workloadManifestProvider =
-                new SdkDirectoryWorkloadManifestProvider(_dotnetPath,
-                    string.IsNullOrWhiteSpace(_targetSdkVersion)
-                        ? currentSdkReleaseVersion.ToString()
-                        : _targetSdkVersion,
-                    _userProfileDir);
+            tempDirPath ??=
+                (string.IsNullOrWhiteSpace(
+                    result.GetValueForOption(WorkloadListCommandParser.TempDirOption))
+                    ? Path.GetTempPath()
+                    : result.GetValueForOption(WorkloadListCommandParser.TempDirOption));
+            string userProfileDir1 = userProfileDir ?? CliFolderPathCalculator.DotnetUserProfileFolderPath;
             DirectoryPath tempPackagesDir =
-                new(Path.Combine(_userProfileDir, "sdk-advertising-temp"));
+                new(Path.Combine(userProfileDir1, "sdk-advertising-temp"));
             NullLogger nullLogger = new NullLogger();
-            _nugetPackageDownloader = nugetPackageDownloader ??
-                                      new NuGetPackageDownloader(tempPackagesDir, null,
-                                          new FirstPartyNuGetPackageSigningVerifier(tempPackagesDir, nullLogger),
-                                          verboseLogger: nullLogger,
-                                          restoreActionConfig: _parseResult.ToRestoreActionConfig());
-            _workloadResolver = workloadResolver ?? WorkloadResolver.Create(workloadManifestProvider, _dotnetPath, currentSdkReleaseVersion.ToString(), _userProfileDir);
-
-            _workloadRecordRepo = workloadRecordRepo ??
-                WorkloadInstallerFactory.GetWorkloadInstaller(reporter, _currentSdkFeatureBand, _workloadResolver, _verbosity, _userProfileDir,
-                VerifySignatures,
-                elevationRequired: false).GetWorkloadInstallationRecordRepository();
+            nugetPackageDownloader ??= new NuGetPackageDownloader(tempPackagesDir, null,
```

Comment:
Just be aware that a large chunk of this code was refactored in 6.0.3xx a few days ago to move thigns like the downloader, etc into the base workload. I don't think any of that have flowed through to 6.0.4xx/main yet.

Reply:
Thank you Jacques for pointing this out. I suppose you are referring to [those changes](https://github.com/dotnet/sdk/commit/94c3deb2300f40bc5b531a08bc6d2c1aebc6dfac#diff-927d9977a1d2302c4b31778c2949410c7afe077ef1ec0b842061241dca72c43fL14-L146) - correct?

Change to `WorkloadListCommand` in my PR were mainly extracting+moving the code, so that workloads enumerating can be reused beyond just the list command. So those two changes should not be hard to merge.

Forbid my lack of knowledge around schedule of merging from servicing branches to main - when is this expected to happen? It would be probably good if I could incorporate your changes before finalizing this PR.

Reply:
The infrastructure will periodically generate code flow PRs, e.g. it should generate a merge from release/6.0.3xx to release/6.0.4xx, which will then generate a merge from release/6.0.4xx to main